nototools/generate_dingbats_html.py

```diff
@@ -385,47 +385,111 @@ def _load_fonts_and_targets(font_data, target_data, data_dir):
-      # ('Wingdings 3', 'wingdng3_codes.txt', 'wng3', omit_list),
-      # ('Zapf Dingbats', 'zapfdingbats_codes.txt', 'zdng', omit_list)
-  ]
-  fonts, targets = _load_fonts_and_targets(font_data, target_data, data_dir)
-  context = 'O%sg'
+def strip_comments_from_file(filename):
+  with open(filename, 'r') as f:
+    for line in f:
+      ix = line.find('#')
+      if ix >= 0:
+        line = line[:ix]
+      line = line.strip()
+      if not line:
+        continue
+      yield line
+
+
+def _read_font_data_from_file(filename):
+  font_data = []
+  for line in strip_comments_from_file(filename):
+    info = line.split(';')
+    while len(info) < 4:
+      info.append(None)
+    font_data.append(tuple(info))
+  return font_data
+
+
+def _read_target_data_from_file(filename):
+  """Target data uses # to indicate a comment to end of line.
+  Comments are stripped, then an empty or blank line is ignored.
+
+  Each line defines a tuple of four values: target name,
+  codelist, preferred font ids, and omitted font ids.
+  A line can also start with either of two directives,
+  !define and !default.
+
+  If a line starts with '!define ' we expect a key followed
+  by '=' and then one or more names separated by space. The
+  names are turned into a list, and entered into a dictionary
+  for the key.  Once defined a key cannot be redefined.
+
+  If a line starts with '!default ' we expect a key of either
+  'prefer' or 'omit' optionally followed by '=' and a list of
+  names to prefer or omit; these will become the default
+  values until the next '!default ' directive.  If there is
+  no '=' the value is reset.  An omitted or empty prefer or
+  omit field will get the fallback, to explicitly request None
+  and override the fallback the field should contain 'None'.
+
+  Normally, a line consists of 2-4 fields separated by ';'.
+  The first two are a target name and a codelist spec.  The
+  third is the preferred font ids separated by space,
+  previously !defined keys can be used here instead of this
+  list and the list defined for that key will be used.
+  The fourth is the omitted font ids separated by space, they
+  are treated similarly.  If the preferred or omit field is
+  missing or empty and a default value for it has been set,
+  that value is used."""
+
+  def add_index_list_or_defined(info, index, fallback, defines):
+    """Extend or update info[index], possibly using defines"""
+    if len(info) <= index:
+      info.append(fallback)
+    elif info[index] != None:
+      item = info[index]
+      if item in defines:
+        items = defines[item]
+      elif item == 'None':
+        items = None
+      elif item:
+        items = item.split()
+      else:
+        items = fallback
+      info[index] = items
+
+  prefer_fallback = None
+  omit_fallback = None
+  defines = {}
+  target_data = []
+  for line in strip_comments_from_file(filename):
+    if line.startswith('!define '):
+      name, rest = line[8:].split('=')
```

Comment:
nit: forgot to mention earlier.   optional. I wonder if the following would be better. 
```
kDefineDirective = '!define '  
if line.startswith(kDefineDirective):
  name, rest = line[len(kDefineDirective)].split('=')
```

The same can be said of  '!default '.